### PR TITLE
Product taxonomies and attributes handling

### LINKED ISSUE #72: How to handle Variable Product attributes with terms withtout translation
**tl;tr**

I have come across the following scenario and need help figuring out what should be the behavior of the plugin in this case.

Scenario:
1. Product attribute with the name "My Attribute" is created. 
1.1. The translation is automatically added to the Polylang Strings Translations page, with the same name for all active languages.
2. The term named "terms1" is added to "My Attribute" in the default language.
3. A variable product titled "My Var Product" is created in the default language with the attribute "My Attribute" used to create variations.
4. A translation of "My Var Product" is created for a second language, named "My Translated Var Product"
4.1. The translation is created with 1 variation and all metas copied except the term slug since the term doesn't have translation. When "My Translated Var Product" is saved/published, Polylang will sync back to the product in default language "My Var Product" and remove the attribute and variation.

Althought he behaiviour is consistent with Polylang own logic, for a new user of WooPoly plugin this may be confusing and doesn't understand that the terms were not translated.

So my question: shall WooPoly create the terms translations with the same name (like for the attribute name - see point 1.1) if a translation is not found? And potentially add some inline js to alert that attribute terms need to be translated?

Alternatively, we can simple add a note to the readme file and FAQ mentioning this as a know issue?

### LINKED ISSUE #127: Text attribute synchronisation
Products\Attributes of type Select and their Term translations work fine.

However for attributes where a Select value list of Terms and translations is not appropriate like types of product codes (ASIN, UPC etc), numeric values or maybe additional dimensions, one would expect to use:  

Products\Attributes of type Text

And the Text values should be simply repeated/copied for each language.

Currently when you add attribute and value to the original language, the attribute is added to the translated product, but with no value (and not editable)
(tested with simple products and variations)

Also reported on the forums:
https://wordpress.org/support/topic/text-attributes-of-simple-product-not-duplicated-in-translations/

### PR DESCRIPTION
**Pull request by Jon007**
This release fixes a number of issues around handling of attributes and translations.
In particular:

1. New translations can now use auto-copy of source language, to help save time translating.
   In future a machine translation will be added.
	 Copy option covers Product Title, Short Description and Long Description.
   Also when creating a new product, any missing Product Categories, Tags and Attributes are copied,
   to avoid unexpected problems which occur if a translation is saved with missing term translations.

2. it is now possible to set up the system to allow different types of product attributes
to be synchronised, translated, or independent in each language. The default options will be:
 - Translation and Synchronization Enabled for Product Attributes
 - Synchronization off for Custom Product Attributes

In this case choose how to set up your product attributes as follows:
 - Translated Attribute?  Add in Products\Attributes and turn on Translation in Polylang at:
        Languages\Settings\Custom Taxonomies 
 - Synchronised Attribute? [eg same value in all languages, eg product code, numeric properties] 
				Add in Products\Attributes and leave Translation turned off in Polyang.
 - Different value in each language? add directly to Product as a Custom Product Attribute

* Enh: synchronisation for Custom Product Attributes and Global Product Attributes can now be 
       turned on and off independently in 
			 Settings\WooPoly, Metas List, Attributes Metas, Custom Product Attributes.
       The fields locker is unlocked for the attribute types which are not synchronized.
* Fix: Global Product Attributes can now be individually configured in Polylang:  
			 When Settings\WooPoly Translation attributes is checked then attributes appear in Polylang:
       Languages\Settings\Custom Taxonomies lists the individual taxonomies 
			 Previously all attributes translation were forced on: now they can be selectively turned on and off.  This means that there is no longer any need to create dummy translations for untranslateable values such as reference codes and numeric fields.  Fixes #127.

CHANGE: new Product Attributes are no longer automatically enabled for translation,
       After creating new Attribute, enable Translation in Polylang if needed by checking:
			 Languages\Settings\Custom Taxonomies 
* Enh: Missing Term Translations are now added by default. Fixes #72
			 Applies to Products\Categories, Products\Tags, Products\Attributes 
       Previously missing term translations caused
* Fixes: #148 WooCommerce3 product_visibility is now a taxonomy not a meta item  
* Fixes: #153 Fields locker doesn't correctly lock Product Attributes of type Select 
* Fixes: #147 When adding new variations, tool should also add the new variation to other languages

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Contributors: hyyan, decarvalhoaa, jonathanmoorebcsorg
 Tags: cms, commerce, e-commerce, e-shop, ecommerce, multilingual, products, shop, woocommerce, polylang, bilingual, international, language, localization, multilanguage, multilingual, translate, translation
 Requires at least: 3.8
-Tested up to: 4.7
+Tested up to: 4.7.5
```
Comment (hyyan) [inline readme.txt]: Patch versions of wordpress won't break the plugin , so let's just use the last wordpress minor version 4.7

File: readme.txt [hunk 2]
```diff
@@ -113,15 +113,68 @@ Just make sure to setup your permalinks , and every thing will be fine , I promi
...
 == Changelog ==
-
 == 1.0.0 ==
 
-* Fix #137 #131 #130 #110 adaptations for wooCommerce 3.0
+Fixes #149 Enable duplication of variable products
+Fixes #165 upsets/crosssells handling in wooCommerce3
+Fixes #159 Attribute Terms synchronization issues
+
+== 0.30.RC2 WooCommerce3 new taxonomies and Attributes fixes ==
```
Comment (hyyan) [inline readme.txt]: Those changes are part of the version 1.0.0 now as they are merged , plus they are not published on wordpress repo , so let's make this, part of the version 1.0.0 changelog , please

File: readme.txt [hunk 3]
```diff
@@ -113,15 +113,68 @@ Just make sure to setup your permalinks , and every thing will be fine , I promi
...
+       After creating new Attribute, enable Translation in Polylang if needed by checking:
+			 Languages\Settings\Custom Taxonomies 
+* Enh: Missing Term Translations are now added by default. Fixes #72
+			 Applies to Products\Categories, Products\Tags, Products\Attributes 
+       Previously missing term translations caused
+* Fixes: #148 WooCommerce3 product_visibility is now a taxonomy not a meta item  
+* Fixes: #153 Fields locker doesn't correctly lock Product Attributes of type Select 
+* Fixes: #147 When adding new variations, tool should also add the new variation to other languages
+
+== 0.30.RC1 WooCommerce3 Fixes==
```
Comment (hyyan) [inline readme.txt]: see the prev comment

File: readme.txt [hunk 4]
```diff
@@ -257,6 +310,6 @@ woocommerce and polylang , please update immediately
 The release includes important fixes and updates for latest version of 
 woocommerce and polylang , please update immediately 
 
-= 0.30.0 =
+= 1.00.0 =
```
Comment (hyyan) [inline readme.txt]: this is a typo , must be 1.0.0

Comment (Jon007) [inline readme.txt]: ok sure, ok to all comments

**Comment by hyyan** [comment]
@Jon007 Thanks for huge effort, amazing work, clean code  🍺

**Pull request merged**